src/test/scala/org/broadinstitute/dsde/workbench/sam/api/MockSamUserDirectives.scala

```diff
@@ -16,26 +15,16 @@ trait MockSamUserDirectives extends SamUserDirectives {
   val user: SamUser
   val newSamUser: Option[SamUser] = None
 
-  val petSAdomain = "\\S+@\\S+\\.iam\\.gserviceaccount\\.com".r
-
-  private def isPetSA(email: String) = {
-    petSAdomain.pattern.matcher(email).matches
-  }
+  private lazy val fakeOidcHeaders =
+    OIDCHeaders(OAuth2BearerToken("dummy token"), user.googleSubjectId.toLeft(user.azureB2CId.get), user.email, user.googleSubjectId)
 
   override def withActiveUser(samRequestContext: SamRequestContext): Directive1[SamUser] = onSuccess {
-    directoryDAO.loadSubjectFromEmail(user.email, samRequestContext).map { maybeUser =>
-      maybeUser.map { _ =>
-        if (isPetSA(user.email.value)) {
-          user.copy(id = WorkbenchUserId("newuser"), email = WorkbenchEmail("newuser@new.com"))
-        } else {
-          user
```

Comment:
This was the main problem with how the mock worked. It returned the `user` as it was statically defined, not fresh information loaded from the directory dao.